js/ui/codeView.js

```diff
@@ -453,6 +457,63 @@ var CodingSession = new Lang.Class({
         return this._toolbox;
     },
 
+    setGrabbed: function(grabbed) {
+        if (this._grabTimeoutId > 0) {
+            Mainloop.source_remove(this._grabTimeoutId);
+            this._grabTimeoutId = 0;
+        }
+
+        if (!this._button)
+            return;
+
+        if (grabbed) {
+            this._button.opacity = 0;
+        } else {
+            // giving some time to finish possible window movement FXs
+            this._grabTimeoutId = Mainloop.timeout_add(300,
+                () => {
+                    this._button.opacity = 255;
+                    return GLib.SOURCE_REMOVE;
+                });
+        }
+    },
+
+    _attachAppToToolbox: function() {
+        this._appParent = this.app.get_parent();
+        this._appParent.remove_child(this.app);
+        this.toolbox.insert_child_at_index(this.app, 0);
+        this.app.set_position(0, 0);
+        // As a child actor we need to ensure that the position is always (0, 0)
+        // to avoid the rendering outside the parent window or with an offset.
+        // We've code to sync both windows so to avoid glitches and actor
+        // movements inside the parent we force the position listening to the
+        // position change event.
+        this._appPositionHandlerId = this.app.connect('notify::position', (actor) => {
```

Comment:
Have you considered using a `ClutterConstraint` to achieve this instead?
I'm also curious to know how this code interacts with the position tracking code we have elsewhere for coding session windows; ideally we would have a single place where we adjust positions.

Reply:
I tried to do avoid the window sync in the existing function, but for some reason that cause some movement glitches, that's because I was using the `notify`.

I didn't know about `ClutterConstraint` before, but now I've used a `ClutterConstraint` to do this and it works like a charm.